### PR TITLE
feat(files): add Viewer Files ressource handler

### LINKED ISSUE #42916: 🍱 Interactive widget: file
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

With https://github.com/nextcloud/server/issues/42550 coming to nextcloud implement a file widget that allows interaction, in case of a file:
* preview rendering
* download

**Additional context**
https://github.com/nextcloud/server/issues/42550

### LINKED ISSUE #42917: 🍱 Interactive widget: folder
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

With https://github.com/nextcloud/server/issues/42550 coming to Nextcloud implement a folder widget that allows interaction, in case of a folder:
* scrolling the folder
* browsing deeper into the folder

**Additional context**
https://github.com/nextcloud/server/issues/42550

### PR DESCRIPTION
**Pull request by skjnldsv**
- Fix https://github.com/nextcloud/server/issues/42916
- Fix https://github.com/nextcloud/server/issues/42917
- [x] Needs https://github.com/nextcloud-libraries/nextcloud-dialogs/pull/1261

https://github.com/nextcloud/server/assets/14975046/1cf125e6-29cf-4ac0-af9e-318eadda3967

### CODE REVIEW

**Review by AndyScherzinger — APPROVED** [review]
🐘

**Comment by juliushaertl** [comment]
Pushed a small addition to fix other viewer components I've tested (text/files_pdfviewer), hope that is fine

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Ok, I see, I added that for text as we there do not load the file until active is set, but I can work around that in text.

File: apps/files/src/views/ReferenceFileWidget.vue [hunk 1]
```diff
@@ -21,50 +21,155 @@
...
+			<p class="widget-file__description">
 				{{ t('files', 'The file could not be found or you do not have permissions to view it. Ask the sender to share it.') }}
 			</p>
-		</div>
+		</span>
 	</div>
+
+	<!-- Live preview if a handler is available -->
+	<component :is="viewerHandler.component"
+		v-else-if="viewerHandler && !failedViewer"
```
Comment (juliushaertl) [inline apps/files/src/views/ReferenceFileWidget.vue:40]:
```suggestion
		v-else-if="interactive && viewerHandler && !failedViewer"
```

To make sure we load the fallback if interactive view was not requested

Comment (skjnldsv) [reply apps/files/src/views/ReferenceFileWidget.vue:40]: On it! :+1:

**Pull request merged**